File: app/src/main/java/ru/samtakoy/features/import_export/utils/UriUtils.java
package ru.samtakoy.features.import_export.utils;

import android.content.ContentResolver;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.MediaStore;
import android.provider.OpenableColumns;

public class UriUtils {


    public static String getFileNameByUri2(Context context, Uri uri) {
        String result = null;
        if (uri.getScheme().equals("content")) {
            Cursor cursor = context.getContentResolver().query(uri, null, null, null, null);
            try {
                if (cursor != null && cursor.moveToFirst()) {
                    result = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME));
                }
            } finally {
                cursor.close();
            }
        }
        if (result == null) {
            result = uri.getPath();
            int cut = result.lastIndexOf('/');
            if (cut != -1) {
                result = result.substring(cut + 1);
            }
        }
        return result;
    }

    public static String getFileNameByUri(ContentResolver contentResolver, Uri uri)
    {
        String fileName="file";//default fileName
        Uri filePathUri = uri;

        if (uri.getScheme().compareTo("file")==0)
        {
            fileName = filePathUri.getLastPathSegment().toString();
        }
        else
        if (uri.getScheme().toString().compareTo("content")==0)
        {
            Cursor cursor = contentResolver.query(uri, null, null, null, null);
            if (cursor.moveToFirst())
            {
                int column_index = cursor.getColumnIndex(MediaStore.Images.Media.DATA);//Instead of "MediaStore.Images.Media.DATA" can be used "_data"
                if(column_index < 0){
                    return "";
                }
                filePathUri = Uri.parse(cursor.getString(column_index));
                fileName = filePathUri.getLastPathSegment().toString();
            }
        }
        else {
            fileName = fileName+filePathUri.getLastPathSegment();
        }
        return fileName;
    }

}

File: app/src/main/java/ru/samtakoy/features/import_export/utils/UriUtils.kt
package ru.samtakoy.features.import_export.utils

import android.content.ContentResolver
import android.net.Uri
import android.provider.MediaStore

object UriUtils {
    @JvmStatic fun getFileNameByUri(contentResolver: ContentResolver, uri: Uri): String {
        var fileName = "file"
        var filePathUri = uri

        if (uri.getScheme()!!.compareTo("file") == 0) {
            fileName = filePathUri.getLastPathSegment().toString()
        } else if (uri.getScheme().toString().compareTo("content") == 0) {
            val cursor = contentResolver.query(uri, null, null, null, null)
            if (cursor!!.moveToFirst()) {
                val column_index =
                    cursor.getColumnIndex(MediaStore.Images.Media.DATA) //Instead of "MediaStore.Images.Media.DATA" can be used "_data"
                if (column_index < 0) {
                    return ""
                }
                filePathUri = Uri.parse(cursor.getString(column_index))
                fileName = filePathUri.getLastPathSegment().toString()
            }
        } else {
            fileName = fileName + filePathUri.getLastPathSegment()
        }
        return fileName
    }
}
